OracleClient.generateTemporaryTableName leads to query failure in case of concurrent tmp table creation

Oracle connector uses `System.nanoTime()` instead of UUID due to identifier name length limitation:
```java
    @Override
    protected String generateTemporaryTableName()
    {
        return "tmp_trino_" + System.nanoTime();
    }
```

This may lead to query failure when users executed CTAS or INSERT at the same time.
```
Error:  io.trino.plugin.oracle.TestOracleTypeMapping.testBooleanType  Time elapsed: 1.823 s  <<< FAILURE!
java.lang.RuntimeException: 
Error executing sql:
CREATE TABLE boolean_typesg80aqvo4oj AS SELECT CAST(CAST(true AS DECIMAL(1, 0)) AS boolean) col_0,
CAST(CAST(false AS DECIMAL(1, 0)) AS boolean) col_1
	at io.trino.testing.sql.TrinoSqlExecutor.execute(TrinoSqlExecutor.java:45)
	at io.trino.testing.sql.TestTable.<init>(TestTable.java:49)
	at io.trino.testing.sql.TestTable.<init>(TestTable.java:42)
	at io.trino.testing.datatype.CreateAsSelectDataSetup.setupTestTable(CreateAsSelectDataSetup.java:46)
	at io.trino.testing.datatype.SqlDataTypeTest.execute(SqlDataTypeTest.java:78)
	at io.trino.testing.datatype.SqlDataTypeTest.execute(SqlDataTypeTest.java:72)
	at io.trino.plugin.oracle.AbstractTestOracleTypeMapping.testBooleanType(AbstractTestOracleTypeMapping.java:593)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.RuntimeException: ORA-00955: name is already used by an existing object

	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:122)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:506)
	at io.trino.testing.sql.TrinoSqlExecutor.execute(TrinoSqlExecutor.java:42)
	... 19 more
	Suppressed: java.lang.Exception: SQL: CREATE TABLE boolean_typesg80aqvo4oj AS SELECT CAST(CAST(true AS DECIMAL(1, 0)) AS boolean) col_0,
CAST(CAST(false AS DECIMAL(1, 0)) AS boolean) col_1
		at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:509)
		... 20 more
...
	Suppressed: java.lang.RuntimeException: Query: CREATE TABLE "TRINO_TEST"."TMP_TRINO_339516199769" ("COL_0" number(1), "COL_1" number(1))
```

https://github.com/trinodb/trino/runs/5401727429

--- a/plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java
+++ b/plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java
@@ -46,6 +46,7 @@ import oracle.jdbc.OracleTypes;
 import javax.inject.Inject;
 
 import java.math.RoundingMode;
+import java.security.SecureRandom;
 import java.sql.Connection;
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
@@ -107,8 +108,10 @@ import static io.trino.spi.type.TinyintType.TINYINT;
 import static io.trino.spi.type.VarbinaryType.VARBINARY;
 import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
 import static io.trino.spi.type.VarcharType.createVarcharType;
+import static java.lang.Character.MAX_RADIX;
 import static java.lang.Float.floatToRawIntBits;
 import static java.lang.Float.intBitsToFloat;
+import static java.lang.Math.abs;
 import static java.lang.Math.floorDiv;
 import static java.lang.Math.floorMod;
 import static java.lang.Math.max;
@@ -123,6 +126,10 @@ public class OracleClient
         extends BaseJdbcClient
 {
     public static final int ORACLE_MAX_LIST_EXPRESSIONS = 1000;
+
+    // Oracle before 12.2 doesn't allow identifiers over 30 characters
+    private static final int ORACLE_MAX_IDENTIFIER_LENGTH = 30;
+    private static final SecureRandom RANDOM = new SecureRandom();
 
     private static final int MAX_BYTES_PER_CHAR = 4;
 
@@ -213,7 +220,8 @@ public class OracleClient
     @Override
     protected String generateTemporaryTableName()
     {
-        return "tmp_trino_" + System.nanoTime();
+        String tableName = "tmp_trino_" + System.nanoTime() + Long.toString(abs(RANDOM.nextLong()), MAX_RADIX);
+        return tableName.substring(0, min(ORACLE_MAX_IDENTIFIER_LENGTH, tableName.length()));
     }
 
     @Override

